### PR TITLE
avm2: Implement Context3D.setRenderToTexture/setRenderToBackBuffer

### PR DESCRIPTION
**Pull request by Aaron1011**
In the process, I fixed a bug where we were clearing the depth
and stencil buffers with the incorrect value.

This makes Fancy Pants World 4 Part 1 playable to completion
(though there are still some rendering issues that need
to be fixed).

### CODE REVIEW

File: core/src/avm2/globals/flash/geom/Matrix3D.as [hunk 1]
```diff
@@ -274,5 +274,8 @@ package flash.geom {
 			return new Matrix3D(this.rawData);
 		}
 
+		public function copyToMatrix3D(other:Matrix3D):void {
+			other.rawData = rawData
```
Comment (Herschel) [inline core/src/avm2/globals/flash/geom/Matrix3D.as:278]: Should this clone the data?

Comment (Aaron1011) [reply core/src/avm2/globals/flash/geom/Matrix3D.as:278]: It goes through the getter, so it automatically copies it.

**Review by Herschel — APPROVED** [review]

File: tests/tests/swfs/avm2/stage3d_texture/Camera3D.as [hunk 2]
```diff
@@ -0,0 +1,710 @@
+package
+{
+	import flash.geom.Matrix3D;
+	import flash.geom.Vector3D;
+	
+	/**
+	 *   A 3D camera using perspective projection
+	 *   @author Jackson Dunstan
+	 */
```
Comment (Herschel) [inline tests/tests/swfs/avm2/stage3d_texture/Camera3D.as:9]: Is this code under a compatible license?

Comment (Aaron1011) [reply tests/tests/swfs/avm2/stage3d_texture/Camera3D.as:9]: This comes from the zip file at https://www.jacksondunstan.com/articles/1998
Unfortunately, there doesn't seem to be an explicit license. However, we're only using this a test - it's not linked into any part of Ruffle, and it's not distributed as part of releases.

Comment (n0samu) [reply tests/tests/swfs/avm2/stage3d_texture/Camera3D.as:9]: I guess we'd need to ask him
https://www.jacksondunstan.com/articles/1655

**Comment by n0samu** [comment]
With this PR, Fillz and Stacko from #10162 are working great, thank you!
However, Hivex and Nambers crash when trying to enter the first level:
`thread 'main' has overflowed its stack`

File: tests/tests/swfs/avm2/stage3d_texture/Camera3D.as (hunk 2 above)
Comment (Herschel) [reply tests/tests/swfs/avm2/stage3d_texture/Camera3D.as:9]: I'm fine with including it, but it'd be nice get an explicit license as we do distribute source on GH releases. Maybe we could toss the author an e-mail?

**Pull request merged**